tensorboard/backend/context.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Request-scoped context."""
+
+from tensorboard.data import auth as auth_lib
```

Comment:
This feels a bit circular - we will end up with `tensorboard.data.provider` depending on `tensorboard.backend.context` which then depends back on `tensorboard.data.auth`.

See also the other comment on where `context` should live, but it would be ideal if there was a linearizable dependency structure between our high-level API packages.  It also feels a bit strange to say that `auth` lives in `data` if it's going to be the one true "auth" field on `RequestContext`?  As opposed to something that signifies it just being "auth for data providers".

(IMO the most obvious way to avoid that would probably be to generalize `auth` beyond just the data providers and make it part of the same top-level package as `context` and then having `tensorboard.data` take a dependency on that, but curious about your thoughts.)

Reply:
(For posterity, #3826 moves `auth` to the top level.)